subsys/bluetooth/host/att.c

```diff
@@ -421,8 +419,10 @@ struct net_buf *bt_att_chan_create_pdu(struct bt_att_chan *chan, uint8_t op,
 	switch (att_op_get_type(op)) {
 	case ATT_RESPONSE:
 	case ATT_CONFIRMATION:
-		/* Use a timeout only when responding/confirming */
-		buf = bt_l2cap_create_pdu_timeout(NULL, 0, BT_ATT_TIMEOUT);
+		/* Use dedicated buffer pool for responses/confirmations as they
+		 * are self generated by the stack and will never be queued.
+		 */
+		buf = bt_l2cap_create_pdu(&att_rsp_pool, 0);
```

Comment:
Shouldn't it still use the timeout? In case of EATT, then you could still run this buffer pool down, right?

Reply:
Right, I will enable the timeout back, regarding EATT you may have more outstanding requests than the pool count but like I commented before, this is not about always responding immediatelly but having a dedicated pool which the application cannot hold on causing the timeout, so we could in theory just drop to a single response buffer.

Reply:
There is something I still don't understand about the reason for the bug.
If the timeout of 30 seconds was reached, was that because a thread with higher priority would always acquire the buffer first? Or because there was a deadlock?

If we have less buffers here than what could potentially be allocated in the system, for example if we drop it to more buffers, could we still end up with the deadlock waiting for the RSP buffer?
Or was the deadlock on the TX semaphore?

Reply:
According to the issue the application is consuming all the buffers which in turns blocks the RX thread for 30 seconds since it cannot acquire a buffer in the meantime. Not sure if the thread scheduling is supposed to be fair in Zephyr, I would imagine that even if there is a higher priority thread consuming the buffer as fast as possible it should give a chance to lower priority threads to resume at some point, but given the issue does not seem to be the case.

Reply:
The issue description says "Presumably". If we are to say that 1 buffer would be enough for all connections (or number of connections with EATT) then we should know if this is the behavior, or if there was a deadlock.
Blocked for 30 seconds sounds like a deadlock to, so I would like to know if waiting for response buffer could still lead to a deadlock or not, since it determines how many buffers this workaround needs.

Reply:
The time blocked would be how long the buffer needs to be unref, we could alternatively make it configurable if that requires really a zero wait.